python/ray/serve/_private/controller.py

```diff
@@ -269,29 +272,19 @@ def record_autoscaling_metrics(
             replica_id, window_avg, send_timestamp
         )
 
-    def record_handle_metrics(
-        self,
-        deployment_id: str,
-        handle_id: str,
-        actor_id: Optional[str],
-        handle_source: DeploymentHandleSource,
-        queued_requests: float,
-        running_requests: Dict[str, float],
-        send_timestamp: float,
-    ):
+    def record_handle_metrics(self, report: HandleMetricReport) -> None:
         logger.debug(
-            f"Received metrics from handle {handle_id} for deployment {deployment_id}: "
-            f"{queued_requests} queued requests and {running_requests} running requests"
-        )
-        self.autoscaling_state_manager.record_request_metrics_for_handle(
-            deployment_id=deployment_id,
-            handle_id=handle_id,
-            actor_id=actor_id,
-            handle_source=handle_source,
-            queued_requests=queued_requests,
-            running_requests=running_requests,
-            send_timestamp=send_timestamp,
+            f"Received metrics from handle {report.handle_id} "
+            f"for deployment {report.deployment_id}: "
+            f"{report.queued_requests} queued requests "
+            f"and {report.running_requests} running requests"
         )
+        self.autoscaling_state_manager.record_request_metrics_for_handle(report)
+
+    def bulk_record_handle_metrics(self, reports: Sequence[HandleMetricReport]) -> None:
```

Comment:
can the type here be more specific ?

```suggestion
    def bulk_record_handle_metrics(self, reports: List[HandleMetricReport]) -> None:
```